File: java/log-levels/src/main/java/LogLevels.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class LogLevels {
    static Pattern regex = Pattern.compile("^\\[(.*)\\]:\\s+(.*\\S+)\\s*$");

    public static String message(String logLine) {
        Matcher matcher = regex.matcher(logLine);
        matcher.matches();

        return matcher.group(2);
    }

    public static String logLevel(String logLine) {

        Matcher matcher = regex.matcher(logLine);
        matcher.matches();

        return matcher.group(1).toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}

File: cpp/log-levels/log_levels.cpp
#include <string>
#include <regex>

using namespace std;


namespace log_line {
    string message(const string& line) {
        regex pattern(R"(\[(\w+)\]:\s*(.*))");
        smatch m;
        return regex_search(line, m, pattern) ? string(m[2]) : "";
    }
    string log_level(const string& line) {
        regex pattern(R"(\[(\w+)\]:)"); 
        smatch m;
        return regex_search(line, m, pattern) ? string(m[1]) : "";
    }
    string reformat(const string& line) {
        return message(line) + " (" + log_level(line) + ")";
    }
}
